crux_time/src/lib.rs

```diff
@@ -44,33 +61,135 @@ where
         Self { context }
     }
 
-    /// Request current time, which will be passed to the app as `chrono::DateTime<Utc>`
+    /// Request current time, which will be passed to the app as a [`TimeResponse`] containing an [`Instant`]
     /// wrapped in the event produced by the `callback`.
     pub fn now<F>(&self, callback: F)
     where
-        F: Fn(DateTime<Utc>) -> Ev + Send + Sync + 'static,
+        F: Fn(TimeResponse) -> Ev + Send + Sync + 'static,
     {
         self.context.spawn({
             let context = self.context.clone();
             async move {
-                let response = context.request_from_shell(TimeRequest).await;
-                let response = response
-                    .0
-                    .parse::<DateTime<Utc>>()
-                    .expect("invalid time format");
-                context.update_app(callback(response));
+                context.update_app(callback(context.request_from_shell(TimeRequest::Now).await));
             }
         });
     }
 
-    /// Request current time, which will be returned as `chrono::DateTime<Utc>`.
+    /// Request current time, which will be passed to the app as a [`TimeResponse`] containing an [`Instant`]
     /// This is an async call to use with [`crux_core::compose::Compose`].
-    pub async fn now_async(&self) -> DateTime<Utc> {
+    pub async fn now_async(&self) -> TimeResponse {
+        self.context.request_from_shell(TimeRequest::Now).await
+    }
+
+    /// Subscribe to receive a notification when the specified [`Instant`] has arrived.
+    pub fn subscribe_instant<F>(&self, instant: Instant, callback: F)
+    where
+        F: Fn(TimeResponse) -> Ev + Send + Sync + 'static,
+    {
+        self.context.spawn({
+            let context = self.context.clone();
+            async move {
+                context.update_app(callback(
+                    context
+                        .request_from_shell(TimeRequest::SubscribeInstant(instant))
+                        .await,
+                ));
+            }
+        });
+    }
+
+    /// Subscribe to receive a notification when the specified [`Instant`] has arrived.
+    /// This is an async call to use with [`crux_core::compose::Compose`].
+    pub async fn subscribe_instant_async(&self, instant: Instant) -> TimeResponse {
         self.context
-            .request_from_shell(TimeRequest)
+            .request_from_shell(TimeRequest::SubscribeInstant(instant))
             .await
-            .0
-            .parse::<DateTime<Utc>>()
-            .expect("invalid time format")
+    }
+
+    /// Subscribe to receive a notification when the specified duration has elapsed.
+    pub fn subscribe_duration<F>(&self, duration: Duration, callback: F)
```

Comment:
Wondering about the name subscribe here.  Maybe I'm just far too down the GraphQL rabbit hole (feel free to call me out if so 😅) but "subscribe" sort of suggests a recurring thing to me.

Seems less ambiguous for `subscribe_instant` because an instant can obviously only happen once - but with duration I'd have to read the docs to see whether I'm going to get an update every `duration` or not.

Reply:
Maybe `notify_after`/`notify_after_duration` or something along those lines?  It's probably not a big deal though if you prefer the current names.

Reply:
Ha! I don't like those names either. I wondered if "wait_for" and "wait_until" work? Although not sure we're exactly waiting. Notify might be a better root

Reply:
The async ones are kind of waiting I guess?  Though yeah, maybe it makes less sense for the callback ones...

Reply:
`notify_`* wins!